packages/service-provider-core/src/connect-info.ts

```diff
@@ -43,11 +120,12 @@ export default function getConnectExtraInfo(
     : null;
   const { serverOs: server_os, serverArch: server_arch } =
     getBuildInfo.getBuildEnv(buildInfo);
+  const isAtlas = !!atlasVersion?.atlasVersion || getBuildInfo.isAtlas(uri);
 
   return {
-    is_atlas: !!atlasVersion?.atlasVersion || getBuildInfo.isAtlas(uri),
-    is_localhost: getBuildInfo.isLocalhost(uri),
-    is_do: getBuildInfo.isDigitalOcean(uri),
+    ...(await getHostInformation(resolvedHostname)),
```

Comment:
This can potentially block for quite some time (it does DNS lookups and HTTP calls), and except for telemetry, we do not currently need this information. 

Can we make this part of the `mongosh:connect` listener that actually emits telemetry instead?